pkg/ottl/ottlfuncs/README.md

```diff
@@ -91,13 +92,94 @@ Examples:
 
 - `delete_matching_keys(resource.attributes, "(?i).*password.*")`
 
+### flatten
+
+`flatten(target, prefix, depth)`
+
+The `flatten` function flattens a `pcommon.Map` by moving items from nested maps to the root. For example, the following map
+
+```json
+{
+  "name": "test",
+  "address": {
+    "street": "first",
+    "house": 1234
+  },
+  "occupants": ["user 1", "user 2"]
+}
+```
+
+is converted to 
+
+```json
+{
+    "name": "test",
+    "address.street": "first",
+    "address.house": 1234,
+    "occupants.0": "user 1",
+    "occupants.1": "user 2"
+}
+```
+
+`target` is a path expression to a `pcommon.Map` type field. `prefix` is an optional string. `depth` is an optional non-negative int.
+
+If `prefix` is supplied, it will be appended to the start of the new keys. This can help you namespace the changes. For example, if in the above example a `prefix` of `app` was configured, the result would be
+
+```json
+{
+    "app.name": "test",
+    "app.address.street": "first",
+    "app.address.house": 1234,
+    "app.occupants.0": "user 1",
+    "app.occupants.1": "user 2"
+}
+```
+
+If `depth` is supplied, the function will only flatten nested maps up to that depth. For example, if a `depth` of `2` was configured, the following map
+
+```json
+{
+  "0": {
+    "1": {
+      "2": {
+        "3": {
+          "4": "value"
+        }
+      }
+    }
+  }
+}
+```
+
+the result would be
+
+```json
+{
+  "0.1.2": {
+    "3": {
+      "4": "value"
+    }
+  }
+}
+```
+
+A `depth` of `0` means that no flattening will occur.
+
+Examples:
```

Comment:
Is it possible to flatten do a depth without a prefix? Something like `flatten(attributes, nil, 2)`. If so what is the syntax? Can we add it in this list of examples?

Reply:
[The syntax for optional parameters, when not using all of them, is to use the name.](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/ottl/LANGUAGE.md#arguments-in-invocations).  So you can do something like `flatten(attributes, depth=2`.  Here it is in action: https://github.com/TylerHelmuth/opentelemetry-collector-contrib/blob/78627a302c104c4c42283f5bc092a7daec794f1c/pkg/ottl/e2e/e2e_test.go#L84

Reply:
TIL you can use the optional parameters like this, thank you

Reply:
Added another example to the README